### ISSUE TITLE
Replace BLE controller

### ISSUE THREAD

**Issue by DoidoYo** [BernardoCovas/ScooterProtocol#1]
Hello,

I want to create a custom controller to replace the BLE controller. Your protocol implementation is able to send throttle and break values, so in theory this is possible correct? Is the protocol the same for ES4 scooters?

Thanks

**Comment by BernardoCovas** [comment]
Hi DoidoYo,

Well, I cannot know if the protocol is the exact same in the ES4 scooter, because I personally don't own one.
However, it is likely that it's very similar, and the only difference is (maybe) the register values.
If you have a working BLE controller, just probe the half-duplex UART from the BLE to ESC, (115200 8n1) and print the bytes to the screen. The BLE controller should be constantly broadcasting the throttle and break values even if the scooter is locked. Only if you turn it off should it stop broadcasting.
This will tell you the throttle/break registers, and the lock/unlock registers, If they are different, you can change them in the code and in theory it can replace the BLE controller.

Cheers

**Comment by eudovic** [comment]
Hi Bernardo!! Did you intercept de yellow wire for interact? as an Y cable?

**Comment by BernardoCovas** [comment]
Hi @eudovic!
So there should be 4 wires connecting the BLE controller to the main controller board, which includes the ESC
These are (if I remember correctly) red, black, yellow and green
The black and red are for power (+5V), the green is connected to the battery pack + terminal (should be at +38V) and the yellow is the half-duplex UART.
However I don't recall precisely which is which, so I recommend using a  multimeter to verify.
You should be able to intercept the traffic by connecting a UART IC to black and yellow

Cheers

**Comment by eudovic** [comment]
Yeah, all this thing are correct for the pins, but my doubt is if i need disconnect for any side(BLE/ESC) or with I can get directly a signal between ble and esc and read on rx arduino port without disconnect of any side as ex.. Can you understand??
I'm trying to intercept breaking this connections anda im reading the start of communication on ble on arduino, after that t I'm trying to pass it to esc for arduino to get a response...  I'm not shure if its a good way or if I can get only jumper the yellow wire to arduino port as ex...
Can you understand the question? Tks so much!

**Comment by BernardoCovas** [comment]
Hi, not entirely sure if I get what you are saying, the grammar is a bit off
But if I got it correctly:
To intercept it, I think you can just connect a jumper to the yellow and black wires, without cutting anything. (Try it out :) )
To send data however, it's more complicated. You can only send data when there is no data flowing already, since this is half duplex. It is easier if you remove the BLE side, but not strictly necessary

also if you're scooter has a working BLE controller, you can actually control it through BLE without this lib

**Comment by eudovic** [comment]
> Hi, not entirely sure if I get what you are saying, the grammar is a bit off
> But if I got it correctly:
> To intercept it, I think you can just connect a jumper to the yellow and black wires, without cutting anything. (Try it out :) )
> To send data however, it's more complicated. You can only send data when there is no data flowing already, since this is half duplex. It is easier if you remove the BLE side, but not strictly necessary
> 
> also if you're scooter has a working BLE controller, you can actually control it through BLE without this lib
Yeah, sorry for the grammar errors!!
It was the question, If i could listen only getting ground and serial (single wire) and reading on some rx  port... Tks for answer!!
